tdas/lista/lista_test/lista_test.go

```diff
@@ -7,38 +7,188 @@ import (
+	lista := ListADT.CrearListaEnlazada[T]()
+	require.True(t, lista.EstaVacia())
+	lista.InsertarPrimero(valor)
+	require.False(t, lista.EstaVacia())
+}
+
+func TestEstaVacia(t *testing.T) {
+	testEstaVacia(t, 1)
+	testEstaVacia(t, "a")
+	testEstaVacia(t, 1.1)
+	testEstaVacia(t, true)
+}
 
+func testInsertarUltimo[T any](t *testing.T, v1, v2, v3 T) {
+	lista := ListADT.CrearListaEnlazada[T]()
+	lista.InsertarUltimo(v1)
+	lista.InsertarUltimo(v2)
+	lista.InsertarUltimo(v3)
+	require.Equal(t, v1, lista.VerPrimero()) // Verifica que el primero sea el primer insertado
+	require.Equal(t, v3, lista.VerUltimo())  // Verifica que el último sea el último insertado
+	require.Equal(t, 3, lista.Largo())       // Verifica que el largo sea 3
 }
 
 func TestInsertarUltimo(t *testing.T) {
+	testInsertarUltimo(t, 1, 2, 3)
+	testInsertarUltimo(t, "a", "b", "c")
+	testInsertarUltimo(t, 1.1, 2.2, 3.3)
+	testInsertarUltimo(t, true, false, true)
+}
+
+func testIterar[T any](t *testing.T, v1, v2, v3 T) {
+	lista := ListADT.CrearListaEnlazada[T]()
+	lista.InsertarUltimo(v1)
+	lista.InsertarUltimo(v2)
+	lista.InsertarUltimo(v3)
+	lista.Iterar(func(x T) bool {
+		require.Contains(t, []T{v1, v2, v3}, x)
+		return true
+	})
+}
 
+func TestIterar(t *testing.T) {
+	testIterar(t, 1, 2, 3)
+	testIterar(t, "a", "b", "c")
+	testIterar(t, 1.1, 2.2, 3.3)
+	testIterar(t, true, false, true)
 }
 
-func TestBorrarPrimero(t *testing.T) {
+func TestInsertarConIterador(t *testing.T) {
+	lista := ListADT.CrearListaEnlazada[int]()
+	iterador := lista.Iterador()
 
+	// Insertar elementos usando el iterador
+	iterador.Insertar(1) // Insertar en la lista vacía
+	iterador.Siguiente() // Moverse al siguiente nodo (ahora está en nil)
+	iterador.Insertar(2) // Insertar 2 antes de 1
+	iterador.Siguiente() // Moverse al siguiente nodo (ahora está en 1)
+	iterador.Insertar(3) // Insertar 3 antes de 2
+
+	require.Equal(t, 1, lista.VerPrimero()) // El primero debe ser 3
+	require.Equal(t, 3, lista.VerUltimo())  // El último debe ser 1
+	require.Equal(t, 3, lista.Largo())      // Debe haber 3 elementos
 }
 
-func TestVerPrimero(t *testing.T) {
+func TestRemoverConIterador(t *testing.T) {
+	lista := ListADT.CrearListaEnlazada[int]()
+	iterador := lista.Iterador()
+
+	// Insertar elementos usando el iterador
+	iterador.Insertar(1)
+	iterador.Insertar(2)
+	iterador.Insertar(3)
 
+	// Eliminar el primer elemento usando el iterador
+	iterador.Borrar()
+	require.Equal(t, 2, iterador.VerActual()) // El iterador debe apuntar al segundo elemento
+	require.Equal(t, 2, lista.VerPrimero())   // El primero ahora debe ser 2
 }
 
-func TestVerUltimo(t *testing.T) {
+func testInvariantInVolume[T int | rune | float64](t *testing.T) {
+	lista := ListADT.CrearListaEnlazada[T]()
+	iterador := lista.Iterador()
+
+	for i := 1; i <= vol; i++ {
+		iterador.Insertar(T(i))
+	}
 
+	for i := vol; i > 0; i-- {
+		require.Equal(t, T(i), lista.VerPrimero())
+		require.Equal(t, T(i), lista.BorrarPrimero())
+	}
```

Comment:
La segunda línea es innecesaria, pues el `BorrarPrimero` de la línea siguiente ya verifica la igualdad.
```suggestion
	for i := vol; i > 0; i-- {
		require.Equal(t, T(i), lista.BorrarPrimero())
	}
```